dashboard/pages/api/streaming.ts

```diff
@@ -103,10 +103,23 @@ export async function getSinks() {
   return sinkList
 }
 
+type SourceWrap = Source & {
+  properties: { [key: string]: string }
+}
+
 export async function getSources() {
-  let sourceList: Source[] = (await api.get("/api/sources")).map(
-    Source.fromJSON
-  )
+  let sourceList: SourceWrap[] = (await api.get("/api/sources"))
+    .map(Source.fromJSON)
+    .map((src: Source) =>
+      Object.defineProperty(src, "properties", {
+        get: function () {
+          return this.with_properties
+        },
+        set: function (value) {
+          this.with_properties = value
+        },
+      })
+    )
```

Comment:
Why not just renaming the occurrence of `properties` to `with_properties` in the dashboard code?

Reply:
There is type checking that `getSources` returns `Relation`, which requires field `properties`.
https://github.com/risingwavelabs/risingwave/blob/349d36fc2c76482ff2d653f1c0377193b57a1e7f/dashboard/pages/api/streaming.ts#L64-L74
https://github.com/risingwavelabs/risingwave/blob/349d36fc2c76482ff2d653f1c0377193b57a1e7f/dashboard/pages/api/streaming.ts#L37-L43

By delegating `with_properties` to `properties` and use `SourceWrap` as type wrapper, it keeps the `Relation` abstract of table/source/.. with minimal modification.

Reply:
Since `src.with_properties` stores an `Object` and assignment of `Object` is actually "reference assignment", if all accesses are `src.with_properties.xxx`, we can simply use `src.properties = src.with_properties` rather than setter and getter. I will check it later.

Reply:
> Why not just renaming the occurrence of properties to with_properties in the dashboard code?

Done.